smart/model.py

```diff
@@ -998,6 +1006,42 @@ def _init_4_7_set_initial_conditions(self):
                     u_cur.vector().set_local(values)
                     u_cur.vector().apply("insert")
 
+        for parameter in self.pc:
+            if parameter.type == ParameterType.from_xdmf:
+                # load the time vec from xdmf file
+                def load_timesteps_from_xdmf(xdmffile):
```

Comment:
Is there a specific reason to not expose this as a general function, rather than as a hidden function within set initial conditions?